workspaces/ui-v2/src/optic-components/pages/diffs/AddEndpointsPage/hooks/useCheckboxState.ts

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { IUndocumentedUrl } from '<src>/optic-components/hooks/diffs/SharedDiffState';
+
+export enum CheckboxState {
+  Checked,
+  NotChecked,
+  Indeterminate,
+}
+
+export const useCheckboxState = (
+  visibleUrls: IUndocumentedUrl[],
+  selectedUrls: Set<string>,
+  setSelectedUrls: React.Dispatch<React.SetStateAction<Set<string>>>
+) => {
+  const checkboxState = visibleUrls.every((url) =>
+    selectedUrls.has(url.path + url.method)
+  )
+    ? CheckboxState.Checked
+    : visibleUrls.every((url) => !selectedUrls.has(url.path + url.method))
+    ? CheckboxState.NotChecked
+    : CheckboxState.Indeterminate;
+
+  const toggleSelectAllCheckbox = () => {
```

Comment:
When we have:
- indeterminate / all checked -> unselect all selected in the current view
- not checked -> select all in the current view 

This means we have to take into account what the current visible urls are (i.e. the checkbox behavior should not add / remove if we've got a search term in here)